common/src/crypto.ts

```diff
@@ -132,11 +135,16 @@ const sortObjKeys = (x: StrObject) =>
       return acc;
     }, {});
 
-const hashJWK = (x: JsonWebKey) =>
-  new Md5().update(JSON.stringify(sortObjKeys(x as StrObject))).toString();
+const hashJWK = (x: JsonWebKey): Sha256 =>
+  new Sha256().update(JSON.stringify(sortObjKeys(x as StrObject)));
 
-export const hashPublicKey = ({ encryption, signing }: PublicKey) =>
-  hashJWK(encryption) + hashJWK(signing);
+export const hashPublicKey = (
+  { encryption, verification }: PublicKey,
+): string =>
```

Comment:
would giving this a type alias (e.g. `type HashedKey = string;`) give more safety?

Reply:
I’m not sure exactly how type aliases work in TS, I’ll check. It could be that it won’t protect against passing in strings that are not HashedKey

Reply:
iiuc if both x and y are type aliases for a string and you have a function that expects x it will show an error if you send it a "y" however if you have a function that accepts a string, you can send either x or y

at least i hope that's how it works :)

Reply:
It does work as you said :)